common/unsafe/src/main/java/org/apache/spark/unsafe/memory/UnsafeMemoryAllocator.java

```diff
@@ -19,15 +19,24 @@
 
 import org.apache.spark.unsafe.Platform;
 
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.nio.ByteBuffer;
+
+import sun.nio.ch.DirectBuffer;
+
 /**
  * A simple {@link MemoryAllocator} that uses {@code Unsafe} to allocate off-heap memory.
  */
 public class UnsafeMemoryAllocator implements MemoryAllocator {
 
   @Override
-  public MemoryBlock allocate(long size) throws OutOfMemoryError {
+  public OffHeapMemoryBlock allocate(long size) throws OutOfMemoryError {
+    // No usage of DirectByteBuffer.allocateDirect is current design
```

Comment:
why mention this here?

Reply:
Since previous implementation used `DirectByteBuffer.allocateDirect`, I leave a reference for my decision. I will remove this.

Reply:
Let me put this comment here
```
No usage of DirectByteBuffer.allocateDirect is current design
Platform.allocateMemory is used here.
http://downloads.typesafe.com/website/presentations/ScalaDaysSF2015/T4_Xin_Performance_Optimization.pdf#page=26
```